### PR TITLE
Added ability to construct Enemy with Weapon

### PR DESCRIPTION
**Pull request by yaboistyryl**
Should be able to construct random enemy with random weapon. Eventually will need balancing when alpha testing begins.

### CODE REVIEW

File: adventureGame/AdventureGame/.spyproject/unittest.ini [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [unittest]
 framework = pytest
-wdir = C:/Users/mr_ep/OneDrive/Documents/University/Software Development/Assignment 2/adventureGameRepo/AdventureGame/adventureGame\AdventureGame
```
Comment (Aymanchoudhery) [inline adventureGame/AdventureGame/.spyproject/unittest.ini]: Change line to:
wdir = C:/Users/mr_ep/OneDrive/Documents/University/Software Development/Assignment 2/adventureGameRepo/AdventureGame/adventureGame/AdventureGame

File: adventureGame/AdventureGame/obj/DungeonRoom.py [hunk 2]
```diff
@@ -19,12 +19,9 @@ class dungeonRoom:
     # bool _hasEnemies - Whether the room has enemies or not.
     # int _enemyNumber - The amount of enemies in the dungeon room.
     # bool _hasChest - Whether the room has a chest or not.
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/DungeonRoom.py]: Change summary as some parameters no longer exist

File: adventureGame/AdventureGame/obj/DungeonRoom.py [hunk 3]
```diff
@@ -177,9 +174,9 @@ def generateRandomEnemyList(self, enemyCount):
     @staticmethod
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/DungeonRoom.py:168]: Comment above function doesn't match functional standard naming convention

File: adventureGame/AdventureGame/obj/Enemy.py [hunk 4]
```diff
@@ -17,21 +18,13 @@ class Enemy:
     # int _healthPoints - The amount of health the enemy has.
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/Enemy.py]: Change summary as some parameters no longer exist

File: adventureGame/AdventureGame/obj/Enemy.py [hunk 5]
```diff
@@ -71,4 +64,38 @@ def setEquippedWeapon(self, _equippedWeapon):
...
+    # @Param:
+    #   int _playerLevel - The level of the player.
+    def setRandomLevel(self, _playerLevel):
+        if _playerLevel - 2 <= 0:
+            self.setLevel(1)
+        elif _playerLevel + 2 >= 30:
+            self.setLevel(30)
+        else:
+            self.setLevel(random.randint(_playerLevel - 2, _playerLevel + 2))
+
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/Enemy.py]: Add summary

File: adventureGame/AdventureGame/obj/Enemy.py [hunk 6]
```diff
@@ -71,4 +64,38 @@ def setEquippedWeapon(self, _equippedWeapon):
...
+        if _playerLevel - 2 <= 0:
+            self.setLevel(1)
+        elif _playerLevel + 2 >= 30:
+            self.setLevel(30)
+        else:
+            self.setLevel(random.randint(_playerLevel - 2, _playerLevel + 2))
+            
+    def setHealthPointsFromLevel(self):
+        self.setHealthPoints(self.level * 10)
+
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/Enemy.py]: Add summary

File: adventureGame/AdventureGame/obj/Weapon.py [hunk 7]
```diff
@@ -14,19 +18,16 @@ class Weapon:
     # int _value - The value of the weapon.
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/Weapon.py]: Updated summary

File: adventureGame/AdventureGame/obj/Weapon.py [hunk 8]
```diff
@@ -59,3 +60,73 @@ def setDefence(self, _defence):
             self.defence = _defence
         else:
             raise TypeError("Weapon defence expected an int. Received: " + str(type(_defence)) + " Check the type")
+
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/Weapon.py:58]: Add summary

File: adventureGame/AdventureGame/obj/Weapon.py [hunk 9]
```diff
@@ -59,3 +60,73 @@ def setDefence(self, _defence):
...
+    def setRandomName(self):#
+        rarities = ["Common", "Uncommon", "Rare", "Ultra Rare", "Legendary"]
+        namesList = ["Sword", "Long Sword", "Great Sword", "Daggers", "Karambit", "Knife", "Sickle", "Spear", "Scythe", "Mace"]
+        
+        randomNameIndex = random.randint(0, len(namesList) - 1)
+        
+        fullName = str(rarities[self.rarity]) + " " + str(namesList[randomNameIndex])
+        
+        self.setName(fullName)
+
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/Weapon.py]: Add summary

File: adventureGame/AdventureGame/obj/Weapon.py [hunk 10]
```diff
@@ -59,3 +60,73 @@ def setDefence(self, _defence):
...
+        
+        fullName = str(rarities[self.rarity]) + " " + str(namesList[randomNameIndex])
+        
+        self.setName(fullName)
+    
+    def setRandomValue(self):
+        baseValue = random.randint(1, 100)
+        
+        self.setValue(baseValue * self.rarityMultiplier[self.rarity])
+
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/Weapon.py:88]: Add summary

File: adventureGame/AdventureGame/obj/Weapon.py [hunk 11]
```diff
@@ -59,3 +60,73 @@ def setDefence(self, _defence):
...
+    def setRandomValue(self):
+        baseValue = random.randint(1, 100)
+        
+        self.setValue(baseValue * self.rarityMultiplier[self.rarity])
+    
+    def setRandomDamage(self):
+        baseDamage = random.randint(1, 10)
+        
+        self.setDamage(baseDamage * self.rarityMultiplier[self.rarity])
+
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/obj/Weapon.py:95]: Add summary

File: adventureGame/AdventureGame/src/Main.py [hunk 12]
```diff
@@ -117,12 +117,9 @@ def constructPlayers(_playerOneName, _playerTwoName):
     player2 = CharacterClass.Character(_playerTwoName, level, healthPoints, manaPoints, experience, defence, inventory, equippedWeapon, armourSet, gold)
```
Comment (yaboistyryl) [inline adventureGame/AdventureGame/src/Main.py]: Add summary

**Review by yaboistyryl — COMMENTED** [review]
Please observe review and make changes. Alex you are still required to do a review

File: adventureGame/AdventureGame/obj/DungeonRoom.py (hunk 2 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/DungeonRoom.py]: Fixed.

File: adventureGame/AdventureGame/obj/Enemy.py (hunk 4 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/Enemy.py]: Fixed

File: adventureGame/AdventureGame/obj/Enemy.py (hunk 6 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/Enemy.py]: Fixed.

File: adventureGame/AdventureGame/obj/Weapon.py (hunk 7 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/Weapon.py]: Fixed.

File: adventureGame/AdventureGame/obj/Weapon.py (hunk 8 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/Weapon.py:58]: Fixed.

File: adventureGame/AdventureGame/obj/Weapon.py (hunk 10 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/Weapon.py:88]: Fixed.

File: adventureGame/AdventureGame/obj/Weapon.py (hunk 9 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/Weapon.py]: Fixed.

File: adventureGame/AdventureGame/obj/Weapon.py (hunk 11 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/Weapon.py:95]: Fixed.

File: adventureGame/AdventureGame/src/Main.py (hunk 12 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/src/Main.py]: Fixed.

File: adventureGame/AdventureGame/obj/Enemy.py (hunk 5 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/Enemy.py]: Fixed.

File: adventureGame/AdventureGame/obj/DungeonRoom.py (hunk 3 above)
Comment (yaboistyryl) [reply adventureGame/AdventureGame/obj/DungeonRoom.py:168]: Fixed.

**Review by luciferkreuger — APPROVED** [review]

**Pull request merged**